bokehjs/render.js

```diff
@@ -0,0 +1,169 @@
+var fs = require("fs");
+var path = require("path");
+var uuid = require("uuid");
+var yargs = require("yargs");
+var jsdom = require("jsdom");
+var htmlparser2 = require("htmlparser2");
+
+var argv = yargs
+  .options({
+    filename: {
+      type: 'string',
+    },
+    l: {
+      alias: 'log-level',
+      type: 'string',
+      choices: ['debug', 'info', 'warn', 'error', 'fatal', 'none'],
+      default: 'info',
+    },
+    js: {
+      type: 'string',
+      default: path.join(__dirname, 'build/js'),
+    },
+    css: {
+      type: 'string',
+      default: path.join(__dirname, 'build/css'),
+    },
+  })
+  .help('h').alias('h', 'help')
+  .argv;
+
+function readStdin() {
+  var stdin = process.stdin;
+  var data = "";
+
+  stdin.resume();
+  stdin.setEncoding('utf8');
+
+  stdin.on('data', function (chunk) {
+    data += chunk;
+  });
+
+  stdin.on('end', function () {
+    var docs_json = JSON.parse(data);
+    render(docs_json, null);
+  });
+}
+
+function readFile(file) {
+  var docs_json;
+
+  switch (path.extname(file)) {
+    case ".html":
```

Comment:
we discussed looking at this PR on the call today so I'm not sure if it's meant to be final yet - if so, again I don't think we should include html-to-png in the final, it doesn't seem very useful and it's an extra thing that can break and needs maintaining.